llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -3110,6 +3110,33 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
     }
     break;
   }
+  case ISD::STEP_VECTOR: {
+    const APInt &Step = Op.getConstantOperandAPInt(0);
+
+    if (Step.isPowerOf2())
+      Known.Zero.setLowBits(Step.logBase2());
+
+    const Function &F = getMachineFunction().getFunction();
+
+    if (!isUIntN(BitWidth, Op.getValueType().getVectorMinNumElements()))
```

Comment:
I think this might be off by 1. Only `Op.getValueType().getVectorMinNumElements() - 1` needs to fit in the bitwidth.

Reply:
Err. nevermind we need it to fit to load it into the APInt.